### PR TITLE
feature/818

### LINKED ISSUE #818: Disallow selection of aircraft who cannot receive instructions
When an aircraft is outside of our radar coverage, it is able to be selected, and the aircraft's callsign is populated in the command bar. However, since the aircraft has not yet entered our airspace, there is no flight strip on them, nor will they respond to any of our commands.

_To prevent users from not realizing the aircraft is not yet controllable, do not allow the aircraft to appear "selected", either in the command bar, or with the enlarged target dot._
image

### PR DESCRIPTION
**Pull request by n8rzz**
Resolves openscope/openscope#818 

disables selection of an aircraft outside controlled airspace

### CODE REVIEW

File: src/assets/scripts/client/canvas/CanvasController.js [hunk 1]
```diff
@@ -841,25 +841,16 @@ export default class CanvasController {
...
-            trailling_length *= round(dpr);
-        }
-
         cc.save();
 
-        let fillStyle = this.theme.RADAR_TARGET.HISTORY_DOT_INSIDE_RANGE;
-
         if (!aircraftModel.inside_ctr) {
-            fillStyle = this.theme.RADAR_TARGET.HISTORY_DOT_OUTSIDE_RANGE;
+            fillStyle = this.theme.RADAR_TARGET.HISTORY_DOT_INSIDE_RANGE;
```
Comment (erikquinn) [inline src/assets/scripts/client/canvas/CanvasController.js]: definitely still want `HISTORY_DOT_OUTSIDE_RANGE` here

File: src/assets/scripts/client/constants/themes/default/modules/dataBlock.js [hunk 2]
```diff
@@ -180,7 +180,7 @@ export const DATA_BLOCK_THEME = {
      * @memberof DATA_BLOCK_THEME
      * @property TEXT_OUT_OF_RANGE
      */
-    TEXT_OUT_OF_RANGE: COLOR.WHITE,
+    TEXT_OUT_OF_RANGE: COLOR.WHITE_05,
```
Comment (erikquinn) [inline src/assets/scripts/client/constants/themes/default/modules/dataBlock.js]: data blocks must be the same color at all times (should be solid)

File: src/assets/scripts/client/constants/themes/default/modules/radarTarget.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ export const RADAR_TARGET_THEME = {
      * @memberof RADAR_TARGET_THEME
      * @property HISTORY_DOT_OUTSIDE_RANGE
      */
-    HISTORY_DOT_OUTSIDE_RANGE: COLOR.WHITE,
+    HISTORY_DOT_OUTSIDE_RANGE: COLOR.WHITE_05,
```
Comment (erikquinn) [inline src/assets/scripts/client/constants/themes/default/modules/radarTarget.js]: History dots must be the same color at all times. The higher transparency `WHITE_05` is a good choice, but we must set it as _both_ the `inside_range` and `outside_range` color.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,12 @@
 ## 6.1.0 (April 1, 2018)
 ---
 ### New Features
+- Add Ottawa Macdonald–Cartier International Airport (Canada) [#108](https://github.com/openscope/openscope/issues/108)
+- Add Leeds Bradford Airport (England, United Kingdom) [#265](https://github.com/openscope/openscope/issues/265)
+- Add Kansas City Int'l (KMCI) [#816](https://github.com/openscope/openscope/issues/816)
+- Add Raleigh-Durham Int'l (KRDU) [#819](https://github.com/openscope/openscope/issues/819)
+- Add Pittsburgh International Airport (KPIT) [#785](https://github.com/openscope/openscope/issues/785)
+- Allows selection of only aircraft within controlled airspace [#818](https://github.com/openscope/openscope/issues/818)
```
Comment (erikquinn) [inline CHANGELOG.md]: Not sure what this is about?

Comment (n8rzz) [reply CHANGELOG.md]: probably just a bad merge due to the updated format of the changelog

File: src/assets/scripts/client/canvas/CanvasController.js (hunk 1 above)
Comment (n8rzz) [reply src/assets/scripts/client/canvas/CanvasController.js]: 🤔 hmm, not sure why I changed that one

File: src/assets/scripts/client/InputController.js [hunk 5]
```diff
@@ -315,14 +310,14 @@ export default class InputController {
...
-            this.$commandInput.val('');
-            this._eventBus.trigger(EVENT.DESELECT_ACTIVE_STRIP_VIEW, {});
+            return this.deselectAircraft();
+        }
+
+        if (!aircraftModel.inside_ctr) {
+            const isWarning = true;
+            const message = `No response from: ${aircraftModel.callsign}, they are outside controlled airspace`
+
+            UiController.ui_log(message, isWarning);
```
[the remark this answers is not in the document]
Comment (n8rzz) [reply src/assets/scripts/client/InputController.js]: well that's just too easy 😉

File: src/assets/scripts/client/canvas/CanvasController.js (hunk 1 above)
Comment (n8rzz) [reply src/assets/scripts/client/canvas/CanvasController.js]: oh, I see I [was supposed to] flipped the logic

File: src/assets/scripts/client/constants/themes/default/modules/radarTarget.js (hunk 3 above)
Comment (n8rzz) [reply src/assets/scripts/client/constants/themes/default/modules/radarTarget.js]: done

**Pull request merged**